vars/createGist.groovy

```diff
@@ -0,0 +1,37 @@
+/** Create a GitHub gist via the Java github-api interface
+ *  See https://github-api.kohsuke.org/apidocs/org/kohsuke/github/GHGist.html
+ *  and
+ *  https://github-api.kohsuke.org/apidocs/org/kohsuke/github/GHGistBuilder.html
+ *  for which methods are available
+ *
+ *  Currently this only supports adding single-file gists, but it could have
+ *  multi-file support added in the future if that is useful.
+*/
+
+import org.kohsuke.github.GHGist
+import org.kohsuke.github.GHGistBuilder
+import org.kohsuke.github.GitHub
+import org.kohsuke.github.GitHubBuilder
+
+def String call(String filename, String content, String credentialsId='ocfbot') {
+    node {
+        withCredentials([usernamePassword(
+            credentialsId: credentialsId,
+            passwordVariable: 'GITHUB_PASSWORD',
+            usernameVariable: 'GITHUB_USERNAME')
+        ]) {
+            final GitHub gh = new GitHubBuilder()
+                .withPassword(env.GITHUB_USERNAME, env.GITHUB_PASSWORD)
```

Comment:
Do we use an access token with the bot? It seems like for this we should limit the bot to only posting comments (though this is a nice-to-have, not really critical)

Reply:
It's the same access token that Jenkins uses for everything. From OP:

>I also had to log in to the ocfbot account and update the access token it's using to talk to GitHub to allow for interacting with gists, since that wasn't a permission originally included.

We could create a separate token for this but it seems kind of overkill.

Reply:
Yeah that's fair.

Reply:
To answer the first part, yes, it's set up as a "personal access token" within the `ocfbot` account, so we could reasonably easily create a separate one if needed, but I think that can also be done in the future if separating out permissions or separating into multiple accounts is something that would be valuable.